traffic_portal/app/src/common/service/utils/ServerUtils.js

```diff
@@ -67,6 +67,24 @@ var ServerUtils = function($window, propertiesModel, userModel) {
 		);
 	};
 
+	// Takes a response object from TO and converts all of its alerts to messages acceptable by the messageModel
+	this.convertResponseToMessages = function(response) {
```

Comment:
pretty sure you don't have to do this. messagemodel takes in an array or messages so all you have to do is

```
					messageModel.setMessages(result.alerts, true);
```

or 

```
messageModel.setMessages(result.data.alerts, true);
```

Reply:
actually, i guess i'm not sure what you're doing. if the server returns 1 alert, TP will show 1. If it returns 5, TP will show 5...

Reply:
it looks like you return 2 alerts, right? one is a success and one is a warning. an alerts array of 2 _should_ work fine...

Reply:
You are correct, when I started writing the method I assumed they were differently structured, and I never revisited. It sounds like it would be a better idea to have that method combine all the alerts with the same level instead .

Reply:
> It sounds like it would be a better idea to have that method combine all the alerts with the same level instead 

Maybe. Or just leave as it is...